admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/ApplicantTypeServiceImpl.java

```diff
@@ -47,7 +56,27 @@ public ResponseDTO getApplicantType(RequestDTO dto) {
 		List<KeyValues<String, Object>> list = dto.getAttributes();
 		Map<String, Object> map = new HashMap<>();
 		for (KeyValues<String, Object> keyValues : list) {
-			map.put(keyValues.getAttribute(), keyValues.getValue());
+			String attribute = keyValues.getAttribute();
+			Object value = keyValues.getValue();
+
+			if ("residenceStatus".equals(attribute)) {
```

Comment:
This should not be handled here instead it should be checked in the mvel script. As of today if the provided values do not match any condition it returns "000"